tools/cadd/cadd.py

```diff
@@ -0,0 +1,81 @@
+    """
+
+    url = f"https://cadd.gs.washington.edu/api/v1.0/{cadd_version}/{chromosome}:{position}"
+    try:
+        response = requests.get(url, timeout = 30)
+        if response.status_code == 200:
+            data = response.json()
+            return data
+        print(f"Error: {response.status_code} - {response.text}")
+        return None
+    except requests.exceptions.Timeout:
+        print("Error: Timeout occurred while trying to reach the server.")
+    except requests.exceptions.RequestException as req_err:
+        print(f"Error: {req_err}")
+    except ValueError:
+        print("Error: Invalid JSON format in response.")
+    return None
+
+def fetch_cadd_scores(cadd_version, chromosome, start, end):
+    """
+    Fetches CADD (Combined Annotation Dependent Depletion) scores for a range of genomic positions.
+
+    :param str cadd_version: Version of the CADD model used, e.g., "v1.3" or "GRCh38-v1.7".
+    :param int chromosome: Chromosome number for the genomic region.
+    :param int start: Genomic start position of the region.
+    :param int end: Genomic end position of the region.
+    :return: A dictionary containing CADD scores and annotations
+    for the specified genomic region, or None if an error occurs.
+    """
+
+    url = f"https://cadd.gs.washington.edu/api/v1.0/{cadd_version}/{chromosome}:{start}-{end}"
+    try:
+        response = requests.get(url, timeout = 30)
+        if response.status_code == 200:
+            data = response.json()
+            return data
+        print(f"Error: {response.status_code} - {response.text}")
+        return None
+    except requests.exceptions.Timeout:
+        print("Error: Timeout occurred while trying to reach the server.")
+    except requests.exceptions.RequestException as req_err:
+        print(f"Error: {req_err}")
+    except ValueError:
+        print("Error: Invalid JSON format in response.")
+    return None
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Fetch CADD scores for genomic positions.")
+    parser.add_argument("version", help="CADD version, e.g., 'v1.3' or 'GRCh38-v1.7'")
+    parser.add_argument("chromosome", type=int, help="Chromosome number")
+    parser.add_argument("--position", type=int, help="Genomic position (for single SNV)")
+    parser.add_argument("--start", type=int,
+                        help="Genomic start position (for a range of positions)")
+    parser.add_argument("--end", type=int, help="Genomic end position (for a range of positions)")
+
+    args = parser.parse_args()
+
+    if args.position:
+        result = fetch_cadd_score(args.version, args.chromosome, args.position)
+        print(result)
+    elif args.start and args.end:
+        result = fetch_cadd_scores(args.version, args.chromosome, args.start, args.end)
+        print(result)
+    else:
+        print("Please provide either '--position' for single SNV \
+              or '--start' and '--end' for a range of positions.")
```

Comment:
In tests you will find pipeline.ipynb where collected data is showed. create new cell and provide example of usage there for whole dataFrame. If not all columns are displayed, search for pandas.options for max_columns